org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/Scanner.java

```diff
@@ -5906,4 +5906,18 @@ protected void growCommentInfoArrays(int length, int newLength) {
 	System.arraycopy(this.commentTagStarts, 0, this.commentTagStarts = new int[newLength], 0, length);
 }
 
+@Override
+public Scanner clone() {
+	return new Scanner(
```

Comment:
I think this is misleading. Scanner has subclasses that now also offer `clone()` but don't return a cloned instance of their subclass. I still prefer Manoj's proposal and inline this expression into CharLiteral and StringLiteral.

Reply:
One alternative would be a copy constructor instead of `clone()` but that would also add new API to Scanner. Not sure it's worth it.

Reply:
No worries, let me revert the previous commit, than we are back at inline creation of separate scanner instances.